src/emcpy/plots/map_tools.py

```diff
@@ -0,0 +1,118 @@
+import cartopy.crs as ccrs
+
+
+class Domain:
+
+    def __init__(self, domain='global'):
+
+        domain = domain.lower()
+
+        map_domains = {
+            "global": self._global,
+            "conus": self._conus,
+            "north america": self._north_america,
+            "europe": self._europe
+            }
+
+        map_domains[domain]()
+
+        try:
+            return map_domains[domain]()
+        except KeyError:
+            raise TypeError(f'{domain} is not a valid domain.' +
+                            'Current domains supported are:\n' +
+                            f'{" | ".join(map_domains.keys())}"')
+
+    def _global(self):
+        """
+        Sets extent, longitude xticks, and latitude yticks
+        for a global domain.
+        """
+        self.extent = (-180, 180, -90, 90)
+        self.xticks = (-180, -120, -60, 0, 60, 120, 180)
+        self.yticks = (-90, -60, -30, 0, 30, 60, 90)
+
+    def _north_america(self):
+        """
+        Sets extent, longitude xticks, and latitude yticks
+        for a north american domain.
+        """
+        self.extent = (-170, -50, 7.5, 75)
+        self.xticks = (-170, -150, -130, -110, -90, -70, -50)
+        self.yticks = (10, 30, 50, 70)
+
+    def _conus(self):
+        """
+        Sets extent, longitude xticks, and latitude yticks
+        for a contiguous United States domain.
+        """
+        self.extent = (-130, -65, 20, 55)
+        self.xticks = (-130, -120, -110, -100, -90, -80, -70, -60)
+        self.yticks = (25, 35, 45, 55)
+
+    def _europe(self):
+        """
+        Sets extent, longitude xticks, and latitude yticks
+        for a European domain.
+        """
+        self.extent = (-12.5, 40, 30, 70)
+        self.xticks = (-10, 0, 10, 20, 30, 40)
+        self.yticks = (30, 40, 50, 60, 70)
+
```

Comment:
```suggestion
    def _custom(self, dd=dict()):
        """
        Sets extent, longitude xticks, and latitude yticks
        for a Custom domain.
        """
        self.extent = dd.extent
        self.xticks = dd.xticks
        self.yticks = dd.yticks
```

Reply:
There needs to be a check here to ensure that extent, xticks, yticks are in the dictionary passed to this. It will fail on `self.extent = dd.extent` with a message like `dict has no key extent` but we should check first and print an error message that is more explicit like `custom domain requires input dictionary with keys extent, xticks, yticks as tuples of size 4` or something like that

Reply:
See my comment above... I think there needs to be a check here to see if all of these exist and are valid values and if not, print a useful error.